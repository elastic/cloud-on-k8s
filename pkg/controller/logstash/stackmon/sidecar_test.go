// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
// or more contributor license agreements. Licensed under the Elastic License 2.0;
// you may not use this file except in compliance with the Elastic License 2.0.

package stackmon

import (
	"context"
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
	corev1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
	logstashv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/logstash/v1alpha1"
	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/defaults"
	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/stackmon/monitoring"
	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
)

func TestWithMonitoring(t *testing.T) {
	sampleLs := logstashv1alpha1.Logstash{
		ObjectMeta: metav1.ObjectMeta{
			Name:      "sample",
			Namespace: "aerospace",
		},
		Spec: logstashv1alpha1.LogstashSpec{
			Version: "8.6.0",
		},
	}
	monitoringEsRef := []commonv1.ObjectSelector{{Name: "monitoring", Namespace: "observability"}}
	logsEsRef := []commonv1.ObjectSelector{{Name: "logs", Namespace: "observability"}}

	fakeMetricsBeatUserSecret := corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{Name: "sample-observability-monitoring-beat-es-mon-user", Namespace: "aerospace"},
		Data:       map[string][]byte{"aerospace-sample-observability-monitoring-beat-es-mon-user": []byte("1234567890")},
	}
	fakeLogsBeatUserSecret := corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{Name: "sample-observability-logs-beat-es-mon-user", Namespace: "aerospace"},
		Data:       map[string][]byte{"aerospace-sample-observability-logs-beat-es-mon-user": []byte("1234567890")},
	}
	fakeEsHTTPCertSecret := corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{Name: "sample-es-http-certs-public", Namespace: "aerospace"},
		Data: map[string][]byte{
			"tls.crt": []byte("7H1515N074r341C3r71F1C473"),
			"ca.crt":  []byte("7H1515N074r341C3r71F1C473"),
		},
	}
	fakeLsHTTPCertSecret := corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{Name: "sample-ls-http-certs-public", Namespace: "aerospace"},
		Data: map[string][]byte{
			"tls.crt": []byte("7H1515N074r341C3r71F1C473"),
			"ca.crt":  []byte("7H1515N074r341C3r71F1C473"),
		},
	}
	fakeClient := k8s.NewFakeClient(&fakeMetricsBeatUserSecret, &fakeLogsBeatUserSecret, &fakeEsHTTPCertSecret, &fakeLsHTTPCertSecret)

	monitoringAssocConf := commonv1.AssociationConf{
		AuthSecretName: "sample-observability-monitoring-beat-es-mon-user",
		AuthSecretKey:  "aerospace-sample-observability-monitoring-beat-es-mon-user",
		CACertProvided: true,
		CASecretName:   "sample-es-monitoring-observability-monitoring-ca",
		URL:            "https://monitoring-es-http.observability.svc:9200",
		Version:        "8.6.0",
	}
	logsAssocConf := commonv1.AssociationConf{
		AuthSecretName: "sample-observability-logs-beat-es-mon-user",
		AuthSecretKey:  "aerospace-sample-observability-logs-beat-es-mon-user",
		CACertProvided: true,
		CASecretName:   "sample-es-logs-observability-monitoring-ca",
		URL:            "https://logs-es-http.observability.svc:9200",
		Version:        "8.6.0",
	}

	tests := []struct {
		name                      string
		ls                        func() logstashv1alpha1.Logstash
		containersLength          int
		esEnvVarsLength           int
		podVolumesLength          int
		metricsVolumeMountsLength int
		logVolumeMountsLength     int
	}{
		{
			name: "without monitoring",
			ls: func() logstashv1alpha1.Logstash {
				return sampleLs
			},
			containersLength: 1,
		},
		{
			name: "with metrics monitoring",
			ls: func() logstashv1alpha1.Logstash {
				sampleLs.Spec.Monitoring.Metrics.ElasticsearchRefs = monitoringEsRef
				monitoring.GetMetricsAssociation(&sampleLs)[0].SetAssociationConf(&monitoringAssocConf)
				return sampleLs
			},
			containersLength:          2,
			esEnvVarsLength:           0,
			podVolumesLength:          3,
			metricsVolumeMountsLength: 3,
		},
		{
			name: "with logs monitoring",
			ls: func() logstashv1alpha1.Logstash {
				sampleLs.Spec.Monitoring.Metrics.ElasticsearchRefs = nil
				sampleLs.Spec.Monitoring.Logs.ElasticsearchRefs = monitoringEsRef
				monitoring.GetLogsAssociation(&sampleLs)[0].SetAssociationConf(&monitoringAssocConf)
				return sampleLs
			},
			containersLength:      2,
			esEnvVarsLength:       1,
			podVolumesLength:      4,
			logVolumeMountsLength: 4,
		},
		{
			name: "with metrics and logs monitoring",
			ls: func() logstashv1alpha1.Logstash {
				sampleLs.Spec.Monitoring.Metrics.ElasticsearchRefs = monitoringEsRef
				monitoring.GetMetricsAssociation(&sampleLs)[0].SetAssociationConf(&monitoringAssocConf)
				sampleLs.Spec.Monitoring.Logs.ElasticsearchRefs = monitoringEsRef
				monitoring.GetLogsAssociation(&sampleLs)[0].SetAssociationConf(&logsAssocConf)
				return sampleLs
			},
			containersLength:          3,
			esEnvVarsLength:           1,
			podVolumesLength:          6,
			metricsVolumeMountsLength: 3,
			logVolumeMountsLength:     4,
		},
		{
			name: "with metrics and logs monitoring with different es ref",
			ls: func() logstashv1alpha1.Logstash {
				sampleLs.Spec.Monitoring.Metrics.ElasticsearchRefs = monitoringEsRef
				monitoring.GetMetricsAssociation(&sampleLs)[0].SetAssociationConf(&monitoringAssocConf)
				sampleLs.Spec.Monitoring.Logs.ElasticsearchRefs = logsEsRef
				monitoring.GetLogsAssociation(&sampleLs)[0].SetAssociationConf(&logsAssocConf)
				return sampleLs
			},
			containersLength:          3,
			esEnvVarsLength:           1,
			podVolumesLength:          7,
			metricsVolumeMountsLength: 3,
			logVolumeMountsLength:     4,
		},
	}

	for _, tc := range tests {
		t.Run(tc.name, func(t *testing.T) {
			ls := tc.ls()
			builder := defaults.NewPodTemplateBuilder(corev1.PodTemplateSpec{}, logstashv1alpha1.LogstashContainerName)
			_, err := WithMonitoring(context.Background(), fakeClient, builder, ls)
			assert.NoError(t, err)

			assert.Equal(t, tc.containersLength, len(builder.PodTemplate.Spec.Containers))
			for _, v := range builder.PodTemplate.Spec.Volumes {
				fmt.Println(v)
			}
			assert.Equal(t, tc.podVolumesLength, len(builder.PodTemplate.Spec.Volumes))

			if monitoring.IsMetricsDefined(&ls) {
				for _, c := range builder.PodTemplate.Spec.Containers {
					if c.Name == "metricbeat" {
						assert.Equal(t, tc.metricsVolumeMountsLength, len(c.VolumeMounts))
					}
				}
			}
			if monitoring.IsLogsDefined(&ls) {
				for _, c := range builder.PodTemplate.Spec.Containers {
					if c.Name == "filebeat" {
						assert.Equal(t, tc.logVolumeMountsLength, len(c.VolumeMounts))
					}
				}
			}
		})
	}
}
