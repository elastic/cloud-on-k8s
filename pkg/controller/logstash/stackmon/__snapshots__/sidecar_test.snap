
[TestMetricbeatConfig/default - 1]
metricbeat.modules:
  - module: logstash
    metricsets:
      - node
      - node_stats
    period: 10s
    hosts: ["https://localhost:9200"]
    xpack.enabled: true
    
    username: elastic
    
    
    password: password
    
    ssl.enabled: true
    # The ssl verification_mode is set to `certificate` in the config template to verify that the certificate is signed by a trusted authority,
    # but does not perform any hostname verification. This is used when SSL is enabled with or without CA, to support self-signed certificate
    # with a custom CA or custom certificates with or without a CA that most likely are not issued for `localhost`.
    ssl.verification_mode: "certificate"
    ssl.certificate_authorities: ["/mount/ca.crt"]

processors:
  - add_cloud_metadata: {}
  - add_host_metadata: {}

# Elasticsearch output configuration is generated

---

[TestMetricbeatConfig/no_password - 1]
metricbeat.modules:
  - module: logstash
    metricsets:
      - node
      - node_stats
    period: 10s
    hosts: ["https://localhost:9200"]
    xpack.enabled: true
    
    username: elastic
    
    
    ssl.enabled: true
    # The ssl verification_mode is set to `certificate` in the config template to verify that the certificate is signed by a trusted authority,
    # but does not perform any hostname verification. This is used when SSL is enabled with or without CA, to support self-signed certificate
    # with a custom CA or custom certificates with or without a CA that most likely are not issued for `localhost`.
    ssl.verification_mode: "certificate"
    ssl.certificate_authorities: ["/mount/ca.crt"]

processors:
  - add_cloud_metadata: {}
  - add_host_metadata: {}

# Elasticsearch output configuration is generated

---

[TestMetricbeatConfig/no_user_+_password - 1]
metricbeat.modules:
  - module: logstash
    metricsets:
      - node
      - node_stats
    period: 10s
    hosts: ["https://localhost:9200"]
    xpack.enabled: true
    
    
    ssl.enabled: true
    # The ssl verification_mode is set to `certificate` in the config template to verify that the certificate is signed by a trusted authority,
    # but does not perform any hostname verification. This is used when SSL is enabled with or without CA, to support self-signed certificate
    # with a custom CA or custom certificates with or without a CA that most likely are not issued for `localhost`.
    ssl.verification_mode: "certificate"
    ssl.certificate_authorities: ["/mount/ca.crt"]

processors:
  - add_cloud_metadata: {}
  - add_host_metadata: {}

# Elasticsearch output configuration is generated

---

[TestMetricbeatConfig/no_TLS - 1]
metricbeat.modules:
  - module: logstash
    metricsets:
      - node
      - node_stats
    period: 10s
    hosts: ["https://localhost:9200"]
    xpack.enabled: true
    
    username: elastic
    
    
    password: password
    
    ssl.enabled: false
    # The ssl verification_mode is set to `certificate` in the config template to verify that the certificate is signed by a trusted authority,
    # but does not perform any hostname verification. This is used when SSL is enabled with or without CA, to support self-signed certificate
    # with a custom CA or custom certificates with or without a CA that most likely are not issued for `localhost`.
    ssl.verification_mode: "certificate"
    ssl.certificate_authorities: ["/mount/ca.crt"]

processors:
  - add_cloud_metadata: {}
  - add_host_metadata: {}

# Elasticsearch output configuration is generated

---

[TestMetricbeatConfig/no_CA - 1]
metricbeat.modules:
  - module: logstash
    metricsets:
      - node
      - node_stats
    period: 10s
    hosts: ["https://localhost:9200"]
    xpack.enabled: true
    
    username: elastic
    
    
    password: password
    
    ssl.enabled: true
    # The ssl verification_mode is set to `certificate` in the config template to verify that the certificate is signed by a trusted authority,
    # but does not perform any hostname verification. This is used when SSL is enabled with or without CA, to support self-signed certificate
    # with a custom CA or custom certificates with or without a CA that most likely are not issued for `localhost`.
    ssl.verification_mode: "certificate"

processors:
  - add_cloud_metadata: {}
  - add_host_metadata: {}

# Elasticsearch output configuration is generated

---

[TestWithMonitoring/without_monitoring - 1]
{
 "metadata": {
  "creationTimestamp": null
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "name": "logstash",
    "resources": {}
   }
  ]
 }
}
---

[TestWithMonitoring/with_metrics_monitoring - 1]
{
 "metadata": {
  "annotations": {
   "logstash.k8s.elastic.co/monitoring-config-hash": "625208580"
  },
  "creationTimestamp": null
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "name": "logstash",
    "resources": {}
   },
   {
    "args": [
     "-c",
     "/etc/metricbeat-config/metricbeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/metricbeat:8.6.0",
    "name": "metricbeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/metricbeat-config",
      "name": "sample-metricbeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/ls-monitoring-association/observability/monitoring/certs",
      "name": "ls-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/metricbeat/data",
      "name": "metricbeat-data"
     }
    ]
   }
  ],
  "volumes": [
   {
    "name": "ls-monitoring-03f555-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-monitoring-observability-monitoring-ca"
    }
   },
   {
    "emptyDir": {},
    "name": "metricbeat-data"
   },
   {
    "name": "sample-metricbeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-ls-monitoring-metricbeat-config"
    }
   }
  ]
 }
}
---

[TestWithMonitoring/with_TLS_metrics_monitoring - 1]
{
 "metadata": {
  "annotations": {
   "logstash.k8s.elastic.co/monitoring-config-hash": "2320787694"
  },
  "creationTimestamp": null
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "name": "logstash",
    "resources": {}
   },
   {
    "args": [
     "-c",
     "/etc/metricbeat-config/metricbeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/metricbeat:8.6.0",
    "name": "metricbeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/metricbeat-config",
      "name": "sample-metricbeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/ls-monitoring-association/observability/monitoring/certs",
      "name": "ls-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/ls-monitoring/aerospace/sample/certs",
      "name": "ls-monitoring-local-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/metricbeat/data",
      "name": "metricbeat-data"
     }
    ]
   }
  ],
  "volumes": [
   {
    "name": "ls-monitoring-03f555-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-monitoring-observability-monitoring-ca"
    }
   },
   {
    "name": "ls-monitoring-local-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-ls-http-certs-public"
    }
   },
   {
    "emptyDir": {},
    "name": "metricbeat-data"
   },
   {
    "name": "sample-metricbeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-ls-monitoring-metricbeat-config"
    }
   }
  ]
 }
}
---

[TestWithMonitoring/with_logs_monitoring - 1]
{
 "metadata": {
  "annotations": {
   "logstash.k8s.elastic.co/monitoring-config-hash": "983125293"
  },
  "creationTimestamp": null
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "env": [
     {
      "name": "LOG_STYLE",
      "value": "file"
     }
    ],
    "name": "logstash",
    "resources": {}
   },
   {
    "args": [
     "-c",
     "/etc/filebeat-config/filebeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/filebeat:8.6.0",
    "name": "filebeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/filebeat-config",
      "name": "sample-filebeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/ls-monitoring-association/observability/monitoring/certs",
      "name": "ls-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/filebeat/data",
      "name": "filebeat-data"
     },
     {
      "mountPath": "/usr/share/logstash/logs",
      "name": "logstash-logs"
     }
    ]
   }
  ],
  "volumes": [
   {
    "emptyDir": {},
    "name": "filebeat-data"
   },
   {
    "name": "ls-monitoring-03f555-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-monitoring-observability-monitoring-ca"
    }
   },
   {
    "name": "sample-filebeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-ls-monitoring-filebeat-config"
    }
   }
  ]
 }
}
---

[TestWithMonitoring/with_metrics_and_logs_monitoring - 1]
{
 "metadata": {
  "annotations": {
   "logstash.k8s.elastic.co/monitoring-config-hash": "808554888"
  },
  "creationTimestamp": null
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "env": [
     {
      "name": "LOG_STYLE",
      "value": "file"
     }
    ],
    "name": "logstash",
    "resources": {}
   },
   {
    "args": [
     "-c",
     "/etc/metricbeat-config/metricbeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/metricbeat:8.6.0",
    "name": "metricbeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/metricbeat-config",
      "name": "sample-metricbeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/ls-monitoring-association/observability/monitoring/certs",
      "name": "ls-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/metricbeat/data",
      "name": "metricbeat-data"
     }
    ]
   },
   {
    "args": [
     "-c",
     "/etc/filebeat-config/filebeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/filebeat:8.6.0",
    "name": "filebeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/filebeat-config",
      "name": "sample-filebeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/ls-monitoring-association/observability/monitoring/certs",
      "name": "ls-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/filebeat/data",
      "name": "filebeat-data"
     },
     {
      "mountPath": "/usr/share/logstash/logs",
      "name": "logstash-logs"
     }
    ]
   }
  ],
  "volumes": [
   {
    "emptyDir": {},
    "name": "filebeat-data"
   },
   {
    "name": "ls-monitoring-03f555-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-logs-observability-monitoring-ca"
    }
   },
   {
    "emptyDir": {},
    "name": "metricbeat-data"
   },
   {
    "name": "sample-filebeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-ls-monitoring-filebeat-config"
    }
   },
   {
    "name": "sample-metricbeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-ls-monitoring-metricbeat-config"
    }
   }
  ]
 }
}
---

[TestWithMonitoring/with_metrics_and_logs_monitoring_with_different_es_ref - 1]
{
 "metadata": {
  "annotations": {
   "logstash.k8s.elastic.co/monitoring-config-hash": "2399876695"
  },
  "creationTimestamp": null
 },
 "spec": {
  "automountServiceAccountToken": false,
  "containers": [
   {
    "env": [
     {
      "name": "LOG_STYLE",
      "value": "file"
     }
    ],
    "name": "logstash",
    "resources": {}
   },
   {
    "args": [
     "-c",
     "/etc/metricbeat-config/metricbeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/metricbeat:8.6.0",
    "name": "metricbeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/metricbeat-config",
      "name": "sample-metricbeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/ls-monitoring-association/observability/monitoring/certs",
      "name": "ls-monitoring-03f555-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/metricbeat/data",
      "name": "metricbeat-data"
     }
    ]
   },
   {
    "args": [
     "-c",
     "/etc/filebeat-config/filebeat.yml",
     "-e"
    ],
    "env": [
     {
      "name": "POD_IP",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "status.podIP"
       }
      }
     },
     {
      "name": "POD_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.name"
       }
      }
     },
     {
      "name": "NODE_NAME",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "spec.nodeName"
       }
      }
     },
     {
      "name": "NAMESPACE",
      "valueFrom": {
       "fieldRef": {
        "apiVersion": "v1",
        "fieldPath": "metadata.namespace"
       }
      }
     }
    ],
    "image": "docker.elastic.co/beats/filebeat:8.6.0",
    "name": "filebeat",
    "resources": {},
    "volumeMounts": [
     {
      "mountPath": "/etc/filebeat-config",
      "name": "sample-filebeat-config",
      "readOnly": true
     },
     {
      "mountPath": "/mnt/elastic-internal/ls-monitoring-association/observability/logs/certs",
      "name": "ls-monitoring-584a4d-ca",
      "readOnly": true
     },
     {
      "mountPath": "/usr/share/filebeat/data",
      "name": "filebeat-data"
     },
     {
      "mountPath": "/usr/share/logstash/logs",
      "name": "logstash-logs"
     }
    ]
   }
  ],
  "volumes": [
   {
    "emptyDir": {},
    "name": "filebeat-data"
   },
   {
    "name": "ls-monitoring-03f555-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-monitoring-observability-monitoring-ca"
    }
   },
   {
    "name": "ls-monitoring-584a4d-ca",
    "secret": {
     "optional": false,
     "secretName": "sample-es-logs-observability-monitoring-ca"
    }
   },
   {
    "emptyDir": {},
    "name": "metricbeat-data"
   },
   {
    "name": "sample-filebeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-ls-monitoring-filebeat-config"
    }
   },
   {
    "name": "sample-metricbeat-config",
    "secret": {
     "optional": false,
     "secretName": "sample-ls-monitoring-metricbeat-config"
    }
   }
  ]
 }
}
---
